File: app/src/main/java/com/pedrosoares/cotacaoapp/core/base/BaseActivity.java
package com.pedrosoares.cotacaoapp.core.base;

import android.content.Context;
import android.net.ConnectivityManager;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity{

    public boolean isConected(){
        boolean conectado;
        ConnectivityManager conectivtyManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        conectado = conectivtyManager.getActiveNetworkInfo() != null
                && conectivtyManager.getActiveNetworkInfo().isConnected();
        return conectado;
    }

}

File: app/src/main/java/com/pedrosoares/cotacaoapp/core/base/BaseActivity.kt
package com.pedrosoares.cotacaoapp.core.base

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity(){

    open fun setFragment(id: Int, fragment: Fragment?, fragmentActivity: FragmentActivity) {
        val fragmentManager = fragmentActivity.supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(id, fragment!!).commit()
    }
}